src/index.ts

```diff
@@ -703,11 +703,51 @@ class ServerlessCustomDomain {
         }
     }
 
+    /**
+     * Gets value by name from cloudformation exports
+     */
+    public async getImportValue(importValueName, nextToken = null): Promise<string> {
+      const params = nextToken ? { NextToken: nextToken } : {};
+      const result = await this.cloudformation.listExports(params).promise();
+      const importValue = result.Exports.find((val: any) => val.Name === importValueName);
+      if (importValue) {
+        return importValue.Value;
+      }
+
+      if (result.NextToken) {
+        return this.getImportValue(importValueName, result.NextToken);
+      }
+
+      return null;
+    }
+
     /**
      * Gets rest API id from CloudFormation stack
      */
     public async getApiId(domain: DomainConfig): Promise<string> {
         if (this.serverless.service.provider.apiGateway && this.serverless.service.provider.apiGateway.restApiId) {
+            if (typeof this.serverless.service.provider.apiGateway.restApiId === "object" &&
+              this.serverless.service.provider.apiGateway.restApiId["Fn::ImportValue"]) {
+              try {
+                const importValueName = this.serverless.service.provider.apiGateway.restApiId["Fn::ImportValue"];
```

Comment:
This might be easier to read if `const importValueName = this.serverless.service.provider.apiGateway.restApiId["Fn::ImportValue"];` is moved above the try/catch and then all references to it use `importValueName`